src/BlockTemplatesController.php

```diff
@@ -138,13 +284,19 @@ public function theme_has_template( $template_name ) {
 	 * @param string $template_name Template to check.
 	 * @return boolean
 	 */
-	public function default_block_template_is_available( $template_name ) {
+	public function block_template_is_available( $template_name ) {
 		if ( ! $template_name ) {
 			return false;
 		}
 
 		return is_readable(
 			$this->templates_directory . '/' . $template_name . '.html'
+		) ||
+		array_filter(
+			$this->get_block_templates(),
+			function ( $template ) use ( $template_name ) {
+				return $template->slug === $template_name;
+			}
```

Comment:
If we only care about templates saved in the database, would this be equivalent?
```suggestion
		) || gutenberg_get_block_template( 'woocommerce//' . $template_name );
```

Not blocking, but I also wonder if this fix should be done [directly in WC core](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-template-loader.php#L110). I think that was an overlook from my part. But the WC core function shouldn't ignore templates saved in the database.

Reply:
@Aljullu you are right, WC Core seems to only look at templates on the filesystem. I can address this later. This has changed during the refactor, do you think we should go straight to the Gutenberg function, or do you think it's OK to use our `get_block_templates` with the `$slugs` arg?